src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -494,45 +496,85 @@ public static void close(Closeable closeable) {
     /**
      * Resolve the specified filename to a URI.
      *
-     * @param filename name os the file
-     * @return resolved header file URI
+     * @param filename name of the file
+     * @return resolved file URI
      * @throws CheckstyleException on failure
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
-        // figure out if this is a File or a URL
+        URI uri = getWebOrFileProtocolUri(filename);
+
+        if (uri == null) {
+            uri = getFilepathOrClasspathUri(filename);
+        }
+
+        return uri;
+    }
+
+    /**
+     * Resolves the specified filename containing 'http', 'https', 'ftp',
+     * and 'file' protocols (or any RFC 2396 compliant URL) to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved file URI or null if URL is malformed or non-existent
+     */
+    public static URI getWebOrFileProtocolUri(String filename) {
         URI uri;
         try {
             final URL url = new URL(filename);
             uri = url.toURI();
         }
-        catch (final URISyntaxException | MalformedURLException ignored) {
+        catch (URISyntaxException | MalformedURLException ignored) {
             uri = null;
         }
+        return uri;
+    }
 
-        if (uri == null) {
+    /**
+     * Resolves the specified local filename, possibly with 'classpath:'
+     * protocol, to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved file URI
+     * @throws CheckstyleException on failure
+     */
+    private static URI getFilepathOrClasspathUri(String filename) throws CheckstyleException {
+        final URI uri;
+        if (new File(filename).exists()) {
             final File file = new File(filename);
-            if (file.exists()) {
-                uri = file.toURI();
-            }
-            else {
-                // check to see if the file is in the classpath
-                try {
-                    final URL configUrl;
-                    if (filename.charAt(0) == '/') {
-                        configUrl = CommonUtil.class.getResource(filename);
-                    }
-                    else {
-                        configUrl = ClassLoader.getSystemResource(filename);
-                    }
-                    if (configUrl == null) {
-                        throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename);
-                    }
-                    uri = configUrl.toURI();
-                }
-                catch (final URISyntaxException ex) {
-                    throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename, ex);
-                }
-            }
+            uri = file.toURI();
+        }
+        else {
+            uri = getResourceFromClassPath(filename.replace(CLASSPATH_URL_PROTOCOL, ""));
```

Comment:
Just pointing out this will rename files in the classpath which have `classpath:` as the file name.

Reply:
This is a good point.  Would I be able to test this and write code to handle this, though?  Since we cannot have such filenames for Windows users as you've pointed out above, I don't know how we could create a test case.

Reply:
I don't remember where, but we have other tests that will skip themselves if the OS is Windows. Don't know if this has transferred through all the changes to tests with new libraries, but it was there at one point.

Reply:
@nmancus1 , please do substr to cut prefix.

Reply:
Done.